pkg/build/builder/docker.go

```diff
@@ -68,9 +68,17 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, initialURLCheckTimeout, os.Stdin, d.gitClient)
 	if err != nil {
-		d.build.Status.Phase = api.BuildPhaseFailed
-		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
-		d.build.Status.Message = api.StatusMessageFetchSourceFailed
+		switch {
+		case strings.Contains(err.Error(), "no such file or directory"):
```

Comment:
and better still would probably be to wrap the error inside fetchSource and check for that explicit type.  that way if fetchsource throws a file not found error for another reason, you won't report the wrong reason.